modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java

```diff
@@ -3949,6 +3957,28 @@ private void checkNearCacheStarted(IgniteCacheProxy<?, ?> cache) throws IgniteCh
         cluster().active(active);
     }
 
+    /** {@inheritDoc} */
+    @Override public void activate() {
+        cluster().state(ClusterState.ACTIVE);
+    }
+
+    /** {@inheritDoc} */
+    @Override public void deactivate(boolean force) {
+        // Check if cluster ir ready for deactivation.
+        if ((cluster().state() == ClusterState.ACTIVE || cluster().state() == ClusterState.ACTIVE_READ_ONLY)
```

Comment:
I suggest:
1. Change condition to `cluster().state() != ClusterState.INACTIVE && !force`
2. Do not execute task. Implement some method like this `ctx.state().isCacheDataCanBeLostOnDeactivation()` which will check attributes and in-memry caches

Reply:
Impelemted